generator/cmd/protoc-gen-gclient/testdata/smprotos/service.proto

```diff
@@ -0,0 +1,108 @@
+option php_namespace = "Google\\Cloud\\SecretManager\\V1";
+option ruby_package = "Google::Cloud::SecretManager::V1";
+
+// Secret Manager Service
+//
+// Manages secrets and operations using those secrets. Implements a REST
+// model with the following objects:
+//
+// * [Secret][google.cloud.secretmanager.v1.Secret]
+// * [SecretVersion][google.cloud.secretmanager.v1.SecretVersion]
+service SecretManagerService {
+  option (google.api.default_host) = "secretmanager.googleapis.com";
+  option (google.api.oauth_scopes) =
+      "https://www.googleapis.com/auth/cloud-platform";
+
+  // Creates a new [Secret][google.cloud.secretmanager.v1.Secret] containing no
+  // [SecretVersions][google.cloud.secretmanager.v1.SecretVersion].
+  rpc CreateSecret(CreateSecretRequest) returns (Secret) {
+    option (google.api.http) = {
+      post: "/v1/{parent=projects/*}/secrets"
+      body: "secret"
+      additional_bindings {
+        post: "/v1/{parent=projects/*/locations/*}/secrets"
+        body: "secret"
+      }
+    };
+    option (google.api.method_signature) = "parent,secret_id,secret";
+  }
+
+   // Gets metadata for a given [Secret][google.cloud.secretmanager.v1.Secret].
+   rpc GetSecret(GetSecretRequest) returns (Secret) {
+    option (google.api.http) = {
+      get: "/v1/{name=projects/*/secrets/*}"
+      additional_bindings { get: "/v1/{name=projects/*/locations/*/secrets/*}" }
+    };
+    option (google.api.method_signature) = "name";
+  }
+}
+// Request message for
+// [SecretManagerService.CreateSecret][google.cloud.secretmanager.v1.SecretManagerService.CreateSecret].
+message CreateSecretRequest {
+  // Required. The resource name of the project to associate with the
+  // [Secret][google.cloud.secretmanager.v1.Secret], in the format `projects/*`
+  // or `projects/*/locations/*`.
+  string parent = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      child_type: "secretmanager.googleapis.com/Secret"
+    }
+  ];
+
+  // Required. This must be unique within the project.
+  //
+  // A secret ID is a string with a maximum length of 255 characters and can
+  // contain uppercase and lowercase letters, numerals, and the hyphen (`-`) and
+  // underscore (`_`) characters.
+  string secret_id = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // Required. A [Secret][google.cloud.secretmanager.v1.Secret] with initial
+  // field values.
+  Secret secret = 3 [(google.api.field_behavior) = REQUIRED];
+}
+
+// Request message for
+// [SecretManagerService.GetSecret][google.cloud.secretmanager.v1.SecretManagerService.GetSecret].
+message GetSecretRequest {
+  // Required. The resource name of the
+  // [Secret][google.cloud.secretmanager.v1.Secret], in the format
+  // `projects/*/secrets/*` or `projects/*/locations/*/secrets/*`.
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "secretmanager.googleapis.com/Secret"
+    }
+  ];
+}
```

Comment:
nit: missing newline at end of file.  We should have a linter for that.

Reply:
done.